### ISSUE TITLE
Update edk2-libc to build with edk2/master and address VS2017/VS2019 compatibility issues (Bugzilla Bug 3549)

### ISSUE THREAD

**Issue by tianocore-issues** [tianocore/edk2-libc#26]
This issue was created automatically with [bugzilla2github](https://github.com/makubacki/bugzilla2github/tree/bz2gh_tianocore)
# Bugzilla Bug 3549
Date: 2021-08-10T13:32:28+00:00
From: Jayaprakash Nevara &lt;<n.jayaprakash>&gt;
To:   @mdkinney
CC:   chicken.oo, @mdkinney

Depends on:   #3361
Last updated: 2021-08-18T20:44:26+00:00

**Comment by tianocore-issues** [comment]
## Comment 16862
Date: 2021-08-10 13:32:28 +0000
From: Jayaprakash Nevara &lt;<n.jayaprakash>&gt;

- Industry Specification: ---
- Release Observed: EDK II Master
- Releases to Fix: EDK II Master
- Target OS: ---
- Bugzilla Assignee(s): @mdkinney

We have ported Python 3.6.8 to run on UEFI using the StdLib available in edk2-libc repo.
We are using the UDK2018 and VS2017 build tool to build Python 3.6.8 interpreter for UEFI.

We have observed that even through UDK2018 supports building applications with VS2017 build tool we have see certain build errors caused due to warnings being treated as errors. These warnings have been ignored in case of the VS2015 compiler by defining the following flags

    MSFT:*_VS2015x86_*_CC_FLAGS       = /Wv:11
    MSFT:*_VS2015xASL_*_CC_FLAGS      = /Wv:11
    MSFT:*_VS2015x86xASL_*_CC_FLAGS   = /Wv:11

Similar flags needs to be applied for VS2017 compiler tool as well.

**Comment by tianocore-issues** [comment]
## Comment 16942
Date: 2021-08-17 11:36:12 +0000
From: @mdkinney

Assign to owner

**Comment by tianocore-issues** [comment]
## Comment 16943
Date: 2021-08-17 11:38:30 +0000
From: @mdkinney

Additional issues need to be addressed to update edk2-libc to 
build with edk2/master.  The latest version of tools_def.txt adds 
the /WHOLEARCHIVE flag that improves the detection of multiply
defined symbols and missing symbols references by C code that
is not always included in the final linked image.

**Comment by tianocore-issues** [comment]
## Comment 16944
Date: 2021-08-17 11:39:07 +0000
From: @mdkinney

Increase the scope of this issues to add support for both
VS2017 and VS2019.

**Comment by tianocore-issues** [comment]
## Comment 16945
Date: 2021-08-17 11:40:55 +0000
From: @mdkinney

Update title and target repos

**Comment by tianocore-issues** [comment]
## Comment 16953
Date: 2021-08-18 01:48:45 +0000
From: @mdkinney

Patch series for review

https://edk2.groups.io/g/devel/message/79458

**Comment by tianocore-issues** [comment]
## Comment 16971
Date: 2021-08-18 18:40:31 +0000
From: @mdkinney

V2 series for review

https://edk2.groups.io/g/devel/message/79527

**Comment by tianocore-issues** [comment]
## Comment 16972
Date: 2021-08-18 18:43:01 +0000
From: @mdkinney

Commits:
https://github.com/tianocore/edk2-libc/commit/0b1a748e73f9eac544a8509b32af82df25dfa397
https://github.com/tianocore/edk2-libc/commit/afa705e3bfcd952031ad8358a8b9ad5f09d6c0cc
https://github.com/tianocore/edk2-libc/commit/91bf254a980832a41a09440bb41ced58c6a4c743
https://github.com/tianocore/edk2-libc/commit/65bae25197fe06a432262dc2da13f654fdb0d715
https://github.com/tianocore/edk2-libc/commit/a13049a64d1589119206e6da6ef683d085d84a32
https://github.com/tianocore/edk2-libc/commit/b652d46a48377e01d3c13380260fa4f593b43eb9
https://github.com/tianocore/edk2-libc/commit/d0885850905bd401dec064510961986423fc3690
https://github.com/tianocore/edk2-libc/commit/bdc5e7d2c9337ef746993e279dd31942f7be0639
https://github.com/tianocore/edk2-libc/commit/735ebf9a114016430d4921134ae16d84545bca92

**Comment by tianocore-issues** [comment]
## Comment 16982
Date: 2021-08-18 20:44:26 +0000
From: @mdkinney

*** Bug 1807 has been marked as a duplicate of this bug. ***
